internal/kernel/scheduler/scheduler.go

```diff
@@ -88,7 +90,11 @@ func (s *Scheduler) Tick(t int64, batchSize int) {
 						}
 					}
 				},
-			})
+			}
+
+			util.Assert(sqe.Callback != nil, "sqe is nil")
```

Comment:
I'm not sure that we need this assertion because we define callback inline so it's guaranteed to be set. We could assert `submission` and `coroutine.metadata` as Gabriel suggested before constructing the sqe